python/paddle/fluid/compiler.py

```diff
@@ -101,6 +116,7 @@ def with_data_parallel(self,
             self
         """
         assert not self._is_data_parallel, "Already compiled with parallel."
+        assert not self._is_inference, "Cannot compile both data parallel and inference"
```

Comment:
The error doc is confusing, it should be `with_data_parallel and with_inference_optimize cannot be used together.`

Reply:
done